hermes/src/store/types.rs

```diff
@@ -0,0 +1,62 @@
+use {
+    super::proof::wormhole_merkle::WormholeMerkleMessageProof,
+    pyth_oracle::{
+        Message,
+        MessageType,
+        PriceFeedMessage,
+    },
+};
+
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct MessageIdentifier {
+    pub price_id: [u8; 32],
+    pub type_:    MessageType,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct ProofSet {
+    pub wormhole_merkle_proof: WormholeMerkleMessageProof,
+}
+
+pub type Slot = u64;
+pub type UnixTimestamp = i64;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum RequestTime {
+    Latest,
+    FirstAfter(UnixTimestamp),
```

Comment:
(Note for later: some people want FirstBefore as well)

Reply:
Daniel, Guillermo, and I had an interesting conversation about this and we believe first after is enough and actually represents the closing price!